src/errors.rs

```diff
@@ -0,0 +1,51 @@
+use actix_web::{body::BoxBody, http::StatusCode, HttpResponse, ResponseError};
+use derive_more::{Display, Error};
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Display, Error)]
+pub enum APIError {
+    NotFound,
+    WrongCredentials,
+    InvalidToken,
+    InternalServer,
```

Comment:
```suggestion
    InternalServerError,
```
Аналогично везде лучше поменять. `NotFound`, ... само за себя говорит, внутренний сервер - нет